src/common/entity/compute_state_display.ts

```diff
@@ -19,6 +19,11 @@ export const computeStateDisplay = (
   }
 
   if (stateObj.attributes.unit_of_measurement) {
+    if (!Number.isNaN(Number(compareState))) {
+      return `${new Intl.NumberFormat(language).format(Number(compareState))} ${
```

Comment:
Not all browsers support this, we should check for support first. 
And break it out into a helper.

There is also a Polyfill btw: <https://formatjs.io/docs/polyfills/intl-numberformat>

Reply:
@bramkragten I added a check for `Intl` and moved it all into a separate file. Let me know if it's not in the right place. I also added a mocha test for the correct thousands separator and decimal character.

I don't think we need the polyfill; browser support is pretty solid already and it will fallback to the raw number which is what we have now.